sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/TungstenAggregationIterator.scala

```diff
@@ -690,6 +697,9 @@ class TungstenAggregationIterator(
         val mapMemory = hashMap.getPeakMemoryUsedBytes
         val sorterMemory = Option(externalSorter).map(_.getPeakMemoryUsedBytes).getOrElse(0L)
         val peakMemory = Math.max(mapMemory, sorterMemory)
+        totalPeakMemory += peakMemory
+        maxPeakMemory += peakMemory
+        totalSpilledSize += TaskContext.get().taskMetrics().memoryBytesSpilled - spilledSizeBefore
```

Comment:
It's possible for pipelined transformations inside of a stage to be executed in separate threads if there's a PythonRDD, RRDD, PipedRDD, or Hive ScriptTransformation transformation in the middle of the chain.